Userland/Libraries/LibCrypto/Checksum/CRC32.cpp

```diff
@@ -11,6 +11,41 @@
 
 namespace Crypto::Checksum {
 
+#if defined(__aarch64__) && defined(__ARM_FEATURE_CRC32)
+
+void CRC32::update(ReadonlyBytes span)
+{
+    // FIXME: Does this require runtime checking on rpi?
+    //        (Maybe the instruction is present on the rpi4 but not on the rpi3?)
```

Comment:
you can check it by using feature detection from the kernel (might be necessary to propagate it to the user space).

you can refer to `Kernel/Random.cpp` to see the api; if youʼre interested in the rpi version, the drill should be the same – see `Kernel/Arch/aarch64/RPi/MMIO.cpp` instead (not really recommended).

Reply:
Do you know if the rpi3 CPU lacks this instruction while rpi4 has it? (If they both have it, then this static version is good enough.)

Reply:
v8.1 enforces the support but none of the rpi-s are apparently adhering to that standard. yet iʼm pretty sure rpi4 has it; rpi3 needs additional check.

you might try to run the qemu in rpi3 mode if you believe it emulates that faithfully—during the boot you should see feature detection results (should also be correct). i try to check scan the net…

Reply:
okay; that seems to be pretty accurate:
— https://openwrt.org/toh/raspberry_pi_foundation/raspberry_pi

boot logs suggest (heavily) that rpi3 misses it, while rpi4 has the hardware support youʼre looking for.

Reply:
Cool, thanks!